packages/components/src/components/common/frameProps.tsx

```diff
@@ -28,20 +40,54 @@ export const withFrameProps = ({ $margin, $maxWidth }: TransientFrameProps) => {
 
         ${$maxWidth &&
         css`
-            max-width: ${$maxWidth};
+            max-width: ${getValueWithUnit($maxWidth)};
+        `};
+        ${$maxHeight &&
+        css`
+            max-height: ${getValueWithUnit($maxHeight)};
+        `};
+        ${$width &&
+        css`
+            width: ${getValueWithUnit($width)};
+        `};
+        ${$height &&
+        css`
+            height: ${getValueWithUnit($height)};
         `};
     `;
 };
 
-export const framePropsStory = {
-    args: {
-        margin: { top: undefined, right: undefined, bottom: undefined, left: undefined },
-    },
-    argTypes: {
-        margin: {
-            table: {
-                category: 'Frame props',
+export const getFramePropsStory = (allowedFrameProps: Array<keyof FrameProps>) => {
+    const argTypes = allowedFrameProps.reduce(
+        (acc, key) => ({
+            ...acc,
+            [key]: {
+                table: {
+                    category: 'Frame props',
+                },
+                type: key === 'margin' ? 'object' : 'number',
```

Comment:
In storybook we primarily show that values are in numbers, not strings, because it would be hard to understand that you need to type `10px` instead of 10. Also it wouldn't be predictable when we use numbers only in margin.

If you have better idea how to do it, let me know